### PR TITLE
Improvements to message quoting

### LINKED ISSUE #3278: bug: Quoted timestamps are wrong
Your Rocket.Chat version: 0.31

Hey folks,

some minor issue. If..

1. Someone posts something (@05.32),
2. Someone quotes the user (@07:43)

the timestamp of the quoted reply changes from 05.32 to 07.43 to match the quoters timestamp, see this example:

borked timestamps

Guess there is a variable mixup in the code.

_Cheers,_
Chris.

### PR DESCRIPTION
**Pull request by lucasvanhalst**
@RocketChat/core 

Closes #3278

* Fixes timestamp being incorrect when quoting a message
* Hide message actions dropdown when clicking the "quote message" button
* Make the timestamp from a quoted message clickable. When clicked it goes to the message.

### CODE REVIEW

File: packages/rocketchat-ui-message/message/message.coffee [hunk 1]
```diff
@@ -5,7 +5,7 @@ Template.message.helpers
 		unless RocketChat.settings.get('UI_DisplayRoles')
 			return []
 		roles = _.union(UserRoles.findOne(this.u?._id)?.roles, RoomRoles.findOne({'u._id': this.u?._id, rid: this.rid })?.roles)
-		return _.compact(_.map(roles, (role) -> return RocketChat.models.Roles.findOne({ _id: role, description: { $exists: 1 } })?.description));
+		return RocketChat.models.Roles.find({_id: {$in: roles}, description: {$exists: true, $ne: ''}}, {_id: false, description: true})
```
Comment (sampaiodiego) [inline packages/rocketchat-ui-message/message/message.coffee]: to restrict fields returned from the query, you need to specify the [fields](http://docs.meteor.com/api/collections.html#fieldspecifiers) property on the second parameter:
```
return RocketChat.models.Roles.find({ _id: { $in: roles }, description: { $exists: true, $ne: '' } }, { fields: { description: true } })
 ```

**Comment by sampaiodiego** [comment]
thanks again for the contribution @lucasvanhalst =)

the function `getPermaLink` you created will not work when you are "sneak peek-ing" a channel because you don't have a subscription for a room once you join it.

**Comment by lucasvanhalst** [comment]
@sampaiodiego hmm I see, but the cog which opens the message actions isn't visible when previewing a channel (it did when I made this, but not anymore on master) so that should never happen

**Comment by sampaiodiego** [comment]
cool.. I thought it was generating the link to render the quoted message (my mistake).. =)

**Pull request merged**